lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyXSDURIResolver.java

```diff
@@ -44,11 +48,39 @@ public class LibertyXSDURIResolver implements URIResolverExtension, IExternalGra
   public static final ResourceToDeploy SERVER_XSD_RESOURCE = new ResourceToDeploy(XSD_RESOURCE_URL,
       XSD_CLASSPATH_LOCATION);
 
+  /**
+   * Will return an existing xsd file or generate one from a Liberty installation if ones exists
+   * 
+   * @param baseLocation
+   * @param publicId
+   * @param systemId
+   * 
+   * @return Path to schema xsd resource
+   */
   public String resolve(String baseLocation, String publicId, String systemId) {
     if (LibertyUtils.isConfigXMLFile(baseLocation)) {
       try {
-        Path serverXSDCacheFile = CacheResourcesManager.getResourceCachePath(SERVER_XSD_RESOURCE);
-        return serverXSDCacheFile.toFile().toURI().toString();
+        String serverXMLUri = URI.create(baseLocation).toString();
+        LibertyWorkspace libertyWorkspace = LibertyProjectsManager.getInstance().getWorkspaceFolder(serverXMLUri);
+
+        if (libertyWorkspace != null) {
+          //Set workspace properties if not set 
+          LibertyUtils.getVersion(serverXMLUri);
+          LibertyUtils.getRuntimeInfo(serverXMLUri);
+
+          //Check workspace for Liberty installation and generate schema.xsd file
+          //Return schema URI as String, otherwise use cached schema.xsd file
+          if (libertyWorkspace.isLibertyInstalled()) {
+            Path schemaGenJarPath = LibertyUtils.findFileInWorkspace(libertyWorkspace, "ws-schemagen.jar");
+            LOGGER.info("Generating schema file from: " + schemaGenJarPath.toString());
+            String serverSchemaUri = generateServerSchemaXsd(libertyWorkspace, schemaGenJarPath);
+            if (serverSchemaUri != null && !serverSchemaUri.isEmpty()) {
+              return serverSchemaUri;
+            }
+          }
+        }
+        Path serverXSDFile = CacheResourcesManager.getResourceCachePath(SERVER_XSD_RESOURCE);
```

Comment:
Should we log that we are using the default schema file and not a generated one here?

Reply:
Added a new log message here too.